tests/regression/test_interpolate_cross_mesh.py

```diff
@@ -0,0 +1,95 @@
+from firedrake import *
+import numpy as np
+import pytest
+
+
+def test_unit_square():
```

Comment:
I think this would be better style to separate the different parts of this test into individual tests. You can use fixtures for the common setup operations.

Reply:
I am in the process of doing something very similar to this. When using fixtures, does it redo the the setup every time you use the fixture? That's been my hesitation thus far.

Reply:
You can set a [fixture's scope](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session) so it only gets created once. I guess that that is tricky when combined with parallel tests.

Reply:
I've done a bit of refactoring to use fixtures, but I think I should probably do more for clarity. I think I'll just have to accept the extra setup cost (given this won't work for parallel tests).